info/custom-vec.md

```diff
@@ -0,0 +1,164 @@
+# Fun with the Rust `Alloc` trait
+
+This post gives a brief tour of Rust's new allocation interface, and some
+experiments with a custom `Vec` implementation.
+
+## Allocation in Rust
+
+Rust has recently made progress in defining an interface for general memory
+allocation. This interface is different from the standard `malloc`/`free`
+interface from C. Here is a shorter version of the [full
+interface](https://doc.rust-lang.org/nightly/alloc/allocator/trait.Alloc.html):
+
+```rust
+pub unsafe trait Alloc {
+  unsafe fn alloc(&mut self, layout: Layout) -> Result<*mut u8, AllocErr>;
+  unsafe fn dealloc(&mut self, ptr: *mut u8, layout: Layout);
+  // alloc and dealloc are sufficient to implement this trait, but there are
+  // several other methods in the full trait.
+}
+```
+
+Where
+[`Layout`](https://doc.rust-lang.org/nightly/alloc/allocator/struct.Layout.html)
+is a type that encapsulates size and alignment information, and
+[`AllocErr`](https://doc.rust-lang.org/nightly/alloc/allocator/enum.AllocErr.html)
+specifies a few common errors that allocators can exhibit (e.g. OOM). The
+biggest difference between this interface and the more traditional one is the
+fact that the caller provides size information during deallocation.
```

Comment:
I'd say that the "biggest differences between this interface and the more traditional one are that allocation requests take an explicit alignment, and size and alignment are provided on deallocation and reallocation."